pyclam/manifold.py

```diff
@@ -232,14 +234,52 @@ def local_fractal_dimension(self) -> float:
         if 'local_fractal_dimension' not in self.cache:
             logging.debug(f'building cache for {self}')
             if self.nsamples == 1:
-                return 0.
-            count = [d <= (self.radius / 2)
-                     for batch in iter(self)
-                     for d in self.distance_from(batch)]
-            count = np.sum(count)
-            self.cache['local_fractal_dimension'] = count if count == 0. else np.log2(len(self.argpoints) / count)
+                self.cache['local_fractal_dimension'] = 1.
+            else:
+                count = sum([
+                    1 if distance <= (self.radius / 2) else 0
+                    for batch in iter(self)
+                    for distance in self.distance_from(batch)
+                ])
+                self.cache['local_fractal_dimension'] = 1. if count == 0 else np.log2(self.cardinality / count)
         return self.cache['local_fractal_dimension']
 
+    @property
+    def ancestors(self) -> List['Cluster']:
+        """ Ancestry of self, excluding self.
+        """
+        return self.manifold.ancestry(self)[:-1]
+
+    @property
+    def descendents(self) -> List['Cluster']:
+        """ All clusters in the subtree starting at self, excluding self.
+        """
+        return [cluster for layer in self.manifold.layers[self.depth + 1:]
+                for cluster in layer.clusters
+                if self.name == cluster.name[:len(self.name)]]
```

Comment:
Doesn't this only get the next layer? And why not just access children? In any event, it looks like docstring != implementation

Reply:
line 258. that's a list slice at the end. next layer to the final layer

Doing this using children would require a recursive approach and takes too long in python.